ui/components/app/whats-new-popup/whats-new-popup.js

```diff
@@ -321,22 +388,26 @@ export default function WhatsNewPopup({ onClose }) {
         {notifications.map(({ id }, index) => {
           const notification = getTranslatedUINotifications(t, locale)[id];
           const isLast = index === notifications.length - 1;
-          // Display the swaps notification with full image
-          // Displays the NFTs & OpenSea notifications 18,19 with full image
-          return index === 0 || id === 1 || id === 18 || id === 19 || id === 21
-            ? renderFirstNotification(
-                notification,
-                idRefMap,
-                history,
-                isLast,
-                trackEvent,
-              )
-            : renderSubsequentNotification(
-                notification,
-                idRefMap,
-                history,
-                isLast,
-              );
+          // Choose the appropriate rendering function based on the id
+          let renderNotification =
+            notificationRenderers[id] || renderSubsequentNotification;
+
+          ///: BEGIN:ONLY_INCLUDE_IN(build-mmi)
+          renderNotification = renderFirstNotification;
+          ///: END:ONLY_INCLUDE_IN
```

Comment:
So... this is an interesting optimization, but I don't think it would pass typescript safety. renderFirstNotification and renderSubsequentNotification have different arguments, so even though the additional parameters for renderFirstNotification are on the end of the function parameters it would fail to compile in typescript due to extra params existing. In order to make this work you should probably change both renderFirst and renderSubsequent take an options bag ({ notification, idRefMap }) instead of positional parameters. When you add jsdoc types to them, you can define a new type using @typedef for the options and pass that type as the type for the first parameter to each function. Even if you don't reference the 4 additional properties on renderFirst in renderSubsequent it would still type check.

Reply:
Yup!